File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        for(int i = 0; i<matrix.length; i++){
            if( search(matrix[i], target) != -1) return true;
        }
        return false;
    }
    public int search(int[] nums, int target) {
        int l = 0, r = nums.length-1;
        while(l<=r){
            int m = l + (r-l)/2 ;
            if(nums[m] == target) return m;
            if(nums[m] < target) l = m+1;
            else r = m-1;
        }
        return -1;
    }
}

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for i in range(len(matrix)):
            if self.search(matrix[i],target) != -1 : return True
        return False
    def search(self, nums: List[int], target: int) -> int:
        l , r = 0, len(nums)-1
        while l <= r:   
            m = (r+l)//2
            if nums[m] == target: return m
            if nums[m] > target : r = m-1
            else : l = m + 1
        return -1